examples/hosts/node-host/src/nodeCodeloader.ts

```diff
@@ -49,27 +50,30 @@ export class NodeCodeLoader {
         const signalPath = `${packageDirectory}/${signalFileName}`;
         const codeEntrypoint = `${packageDirectory}/node_modules/${scope}/${name}`;
 
+        await this.waitForPackageFiles(packageDirectory, signalFileName, this.waitTimeoutMSec);
```

Comment:
If you always wait for signal file creation, the first install will always wait for the timeout to finish. Thats not a desired behavior. You can always try to install if the signal file is not present. Let's say everything was successful but signal file was not generated. It will just try to reinstall again which is okay.